src/main/java/org/jabref/logic/citationkeypattern/BracketedPattern.java

```diff
@@ -420,14 +396,50 @@ public static String getFieldValue(BibEntry entry, String value, Character keywo
                 return sb.toString();
             } else {
                 // we haven't seen any special demands
-                return entry.getResolvedFieldOrAlias(FieldFactory.parseField(val), database).orElse("");
+                return entry.getResolvedFieldOrAlias(FieldFactory.parseField(pattern), database).orElse("");
             }
         } catch (NullPointerException ex) {
             LOGGER.debug("Problem making expanding bracketed expression", ex);
             return "";
         }
     }
 
+    /**
+     * Parses the provided string to an {@link AuthorList}, which are then formatted by {@link LatexToUnicodeAdapter}.
+     * Afterward, any institutions are formatted into an institution key.
+     * @param unparsedAuthors a string representation of authors or editors
+     * @return an {@link AuthorList} consisting of authors and institution keys with resolved latex.
+     */
+    private static AuthorList createAuthorList(String unparsedAuthors) {
+        AuthorList authorList = new AuthorList();
+        for (Author author : AuthorList.parse(unparsedAuthors).getAuthors()) {
+            // If the author is an institution, use an institution key instead of the full name
+            String lastName = author.getLast()
+                                    .map(LatexToUnicodeAdapter::format)
+                                    .map(isInstitution(author) ?
+                                            BracketedPattern::generateInstitutionKey : Function.identity())
+                                    .orElse(null);
+            authorList.addAuthor(
+                    author.getFirst().map(LatexToUnicodeAdapter::format).orElse(null),
+                    author.getFirstAbbr().map(LatexToUnicodeAdapter::format).orElse(null),
+                    author.getVon().map(LatexToUnicodeAdapter::format).orElse(null),
+                    lastName,
+                    author.getJr().map(LatexToUnicodeAdapter::format).orElse(null)
+            );
+        }
+        return authorList;
+    }
+
+    /**
+     * Checks if an author is an institution by verifying that only the last name is present.
+     * @param author the checked author
```

Comment:
A reformat of the file should add an empty line before, shouldn't it?

Reply:
Yes. I had not reformatted the file. I have now done so and tried to address the points where the reformatting is wrong (e.g., use ordered/unordered lists where appropriate instead of the auto-generated `<p>`). I tried fixing some of the additional warnings the file were giving as well.